pkg/adaptor/hypervisor/vsphere/vsphere.go

```diff
@@ -76,12 +79,55 @@ func soapKeepAliveHandler(ctx context.Context, c *vim25.Client) func() error {
 
 func DeleteGovmomiClient(gclient *govmomi.Client) {
 
+	glock.Lock()
+
+	defer glock.Unlock()
+
 	err := gclient.SessionManager.Logout(context.Background())
 	if err != nil {
 		logger.Printf("Vcenter logout failed error: %s", err)
 	}
 }
 
+func CheckSessionWithRestore(ctx context.Context, vmcfg *Config, gclient *govmomi.Client) error {
+
+	glock.Lock()
+
+	defer glock.Unlock()
+
+	active, err := gclient.SessionManager.SessionIsActive(ctx)
+
+	if active {
+		if err == nil {
+			us, _ := gclient.SessionManager.UserSession(ctx)
+			logger.Printf("Authenticated current session key %s for user %s", us.Key, us.UserName)
+			return nil
+		}
+	}
+
+	if err != nil {
+		logger.Printf("Attempting to create new sesssion due to current session error: %s", err)
+	}
+
+	_ = gclient.SessionManager.Logout(ctx) // Cleanup purposes
+
+	urlinfo, _ := soap.ParseURL(vmcfg.VcenterURL)
+
+	urlinfo.User = url.UserPassword(vmcfg.UserName, vmcfg.Password)
+
+	err = gclient.SessionManager.Login(ctx, urlinfo.User)
+	if err != nil {
+		logger.Printf("Vcenter login failed error: %s", err)
+		return err
+	}
+
+	us, _ := gclient.SessionManager.UserSession(ctx)
+
+	logger.Printf("Created new authenticated session key %s for user %s", us.Key, us.UserName)
+
```

Comment:
same as the above comment - https://github.com/confidential-containers/cloud-api-adaptor/pull/332/files#r1012495174

Reply:
Nope we don't